### PR TITLE
Create an SqlOp base class 

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

File: src/frontends/sql/ops/sql_op.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#define SRC_FRONTENDS_SQL_OPS_SQL_OP_H_
+
+#include <optional>
+
+#include "src/frontends/sql/ops/op_traits.h"
+#include "src/ir/ir_context.h"
+#include "src/ir/module.h"
+#include "src/ir/operator.h"
+
+namespace raksha::frontends::sql {
```
[the remark this answers is not in the document]
Comment (markww) [reply src/frontends/sql/ops/sql_op.h:26]: I imagine this is similar to the `::attributes` cases, so feel free to ignore.

[the remark this answers is not in the document]
Comment (bgogul) [reply src/frontends/sql/ops/sql_op.h:26]: I thought about adding a `ops` namespace as well , but decided to keep it within `sql` namespace for now.  Do you see any major advantage? If not, let us keep it simple for now?

File: src/frontends/sql/ops/sql_op.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// Base class for all SQL-related operators.
+class SqlOp : public ir::Operation {
+ public:
+  // Returns the operator associated with this type `T`.
+  template <typename T>
+  static const ir::Operator& GetOperator(const ir::IRContext& context) {
+    absl::string_view operator_name = OpTraits<T>::kName;
+    CHECK(context.IsRegisteredOperator(operator_name))
+        << "SqlOp '" << operator_name << "' is not registered.";
+    return context.GetOperator(operator_name);
```
[the remark this answers is not in the document]
Comment (bgogul) [reply src/frontends/sql/ops/sql_op.h:37]: No, it is not checked in `GetOperator`, which I was surprised about.

[the remark this answers is not in the document]
Comment (markww) [reply src/frontends/sql/ops/sql_op.h:37]: It's not checked directly within `GetOperator`, but it is within `GetNode`, which `GetOperator` delegates to. Granted, it uses the result of the lower level `map.find` to perform this check, but it's the same thing semantically. Still, you can keep it around if you want the extra defensive code.

[the remark this answers is not in the document]
Comment (bgogul) [reply src/frontends/sql/ops/sql_op.h:37]: Ah got it. Let us leave this here, but will stop adding such checks in future.

**Pull request closed without merging**